src/main/kotlin/org/elm/workspace/compiler/ElmBuildAction.kt

```diff
@@ -0,0 +1,81 @@
+package org.elm.workspace.compiler
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.util.messages.Topic
+import org.elm.lang.core.lookup.ClientLocation
+import org.elm.lang.core.lookup.ElmLookup
+import org.elm.lang.core.psi.ElmNamedElement
+import org.elm.lang.core.types.TyUnion
+import org.elm.lang.core.types.findInference
+import org.elm.openapiext.saveAllDocuments
+import org.elm.workspace.ElmProject
+import org.elm.workspace.elmToolchain
+import org.elm.workspace.elmWorkspace
+
+class ElmBuildAction : AnAction() {
+
+    private val elmMainTypes: Set<Pair<String, String>> = hashSetOf(Pair("Platform", "Program"), Pair("Html", "Html"))
+
+    private val elmJsonReport = ElmJsonReport()
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project
+                ?: return
+        saveAllDocuments()
+
+        val elmCLI = project.elmToolchain.elmCLI
+        if (elmCLI == null) {
+            Messages.showErrorDialog("No path to the Elm compiler", "Build Error")
+            return
+        }
+
+        val elmProject = project.elmWorkspace.allProjects.first()
```

Comment:
Please add a TODO comment to handle the multi-project case. 

At a minimum we should use the `DataContext` to get the current editor/document/psifile, and then use `org.elm.workspace.ElmWorkspaceService#findProjectForFile` to determine the "active" Elm project.

But it might be better to just show a picker/popover UI listing all the Elm projects and letting the user pick.